tests/keithley_2700.py

```diff
@@ -0,0 +1,168 @@
+
+
+class Keithley_2700Tests(unittest.TestCase):
+    """
+    Tests for the Keithley2700.
+    """
+
+    def setUp(self):
+        self._lewis, self._ioc = get_running_lewis_and_ioc("keithley_2700", DEVICE_PREFIX)
+        self.ca = ChannelAccess(default_timeout=30, device_prefix=DEVICE_PREFIX)
+        self.ca.wait_for("IDN")
+
+    def test_GIVEN_scan_state_set_WHEN_read_THEN_scan_state_is_as_expected(self):
+        sample_data = [0, 1]
+        expected_state = ["INT", "NONE"]
+        for set_value, expected_value in zip(sample_data, expected_state):
+            self.ca.assert_setting_setpoint_sets_readback(set_value, "SCAN:STATE", "SCAN:STATE:SP", expected_value)
+
+    def test_GIVEN_idn_defined_WHEN_read_THEN_idn_is_as_expected(self):
+        self.ca.assert_that_pv_is("IDN", "Keithley 2700 Emulator IDN")
+
+    def test_GIVEN_delay_state_set_WHEN_read_THEN_delay_state_is_as_expected(self):
+        sample_data = [0, 1]
+        expected_state = ["OFF", "ON"]
+        for set_value, expected_value in zip(sample_data, expected_state):
+            self.ca.assert_setting_setpoint_sets_readback(set_value, "DELAYMODE", "DELAYMODE:SP", expected_value)
+
+    def test_GIVEN_source_set_WHEN_read_THEN_source_is_as_expected(self):
+        sample_data = [0, 1, 2, 3, 4]
+        expected_channel = ["IMM", "TIM", "MAN", "BUS", "EXT"]
+        for set_value, expected_value in zip(sample_data, expected_channel):
+            self.ca.assert_setting_setpoint_sets_readback(set_value, "CONTROLSOURCE", "CONTROLSOURCE:SP",
+                                                          expected_value)
+
+    def test_GIVEN_buffer_size_set_WHEN_read_THEN_buffer_size_is_as_expected_AND_within_range(self):
+        expected_buffer_size = [5500, 0, 2, 70000]
+        alarm_state = [ChannelAccess.ALARM_NONE, ChannelAccess.ALARM_MAJOR, ChannelAccess.ALARM_NONE,
+                       ChannelAccess.ALARM_MAJOR]
+        for set_value, expected_alarm in zip(expected_buffer_size, alarm_state):
+            self.ca.assert_setting_setpoint_sets_readback(set_value, "BUFFER:SIZE", "BUFFER:SIZE:SP",
+                                                          set_value, expected_alarm=expected_alarm)
+
+    def test_GIVEN_buffer_feed_set_WHEN_read_THEN_buffer_feed_is_as_expected(self):
+        sample_data = [0, 1, 2]
+        expected_channel = ["SENS", "CALC", "NONE"]
+        for set_value, expected_value in zip(sample_data, expected_channel):
+            self.ca.assert_setting_setpoint_sets_readback(set_value, "BUFFER:SOURCE", "BUFFER:SOURCE:SP",
+                                                          expected_value)
+
+    def test_GIVEN_init_state_set_WHEN_read_THEN_init_state_is_as_expected(self):
+        sample_data = [0, 1]
+        expected_state = ["ON", "OFF"]
+        for set_value, expected_value in zip(sample_data, expected_state):
+            self.ca.assert_setting_setpoint_sets_readback(set_value, "INITMODE", "INITMODE:SP", expected_value)
```

Comment:
Does this method not assume the structure of the set point (i.e. `<prefix>:SP`) PV if unspecified? If so, please remove the superfluous argument. Apply throughout

Reply:
Removed superfluous arguments